### DISCUSSION TITLE
Cachet 3.x Announcement

### DISCUSSION THREAD

**Discussion by jbrooksuk** [cachethq/cachet#4342] [Announcements]
Hey everyone,

It’s been a while 👋🏻

I want to start by saying that I owe so much to Cachet. My career at [Laravel](https://laravel.com/) being just one of the many things I'm grateful for.

9 years ago, I released Cachet while working on another SaaS application. Cachet became my primary focus and it reached many heights. Fortune 50 companies were using it, it reached #1 on Hacker News, topped Product Hunt and casually sat in GitHub's trending projects for weeks like it was no big deal.

5 years ago, I sold Cachet. The buyers and I had big plans for the project. Unfortunately, those plans never came to life. I personally hold myself responsible for that. Within just a few weeks, my brother passed away and my first daughter was born. After that, I never felt right working on Cachet.

Fast forward 5 years and things have changed dramatically. I'm now in a much better place personally, I love my job at Laravel, I've launched a new product, [Checkmango](https://checkmango.com/) and I have a deep, burning desire to get my baby back.

A few months ago I mentioned in an issue that I was speaking with the owners to see if I could get Cachet back. It's taken us some time but we're now at a point where I have something to announce.

**Cachet, is once again, mine 🎉**

Immediately I have thoughts… Maybe I'm too late now? Maybe too much time has passed? Cachet may've blazed the way for open-source status page systems, but who's even using it anymore?

Honestly, I don’t think that’s a problem. I still get several emails a week about Cachet and the website + repository receive thousands of visitors a month. I firmly believe that we're in a good position to get Cachet going again.

So what's the plan, James?

Looking through the codebase is quite painful... I'm going to rebuild Cachet with Vue.js, Inertia, Laravel 10.x and Tailwind.
The current 2.4 branch of Cachet is **old**. It's easier to rebuild than it is to upgrade at this point. As much as I want 3.x to be 100% compatible, there are some things we need to keep in mind:

1. We'll only support PHP 8.1 and above. Anything below this is now EOL and not supported by Laravel 10.x (which we'll be using).
2. I *may* drop non-English translations. We can bring these back later if that happens.
3. I'm going to do my best to keep all existing installations working. Even if the upgrade takes a bit of work, I want all existing data to continue to live. This one is non-negotiable.

This isn't going to be an overnight job. It's going to take me a while and you'll be able to follow along too. Shortly, I'm going to be cleaning up the repository:

* Closing all issues (I'm going to make a note of any features / fixes I want to include in the new version).
* Closing all pull requests. They'll be incompatible anyway.
* Discussions will remain open.
* Deleting old branches.
* I'm going to be working in a new 3.x branch.

I’ve put real money into making this happen. This is happening. 

Let’s make this super collaborative. Can you help? I’d really appreciate it if you could help me re-build Cachet. Here's what I'm looking for:

* People who are familiar with Tailwind and can help build a new frontend for both the status page and dashboard.
* People who are willing to test Cachet in the future.
* People who can write documentation. There will be a new `cachethq/docs` repository with VitePress installed and ready for the old docs to be migrated. 
* Sponsorship. I'll be honest, this isn't a cheap endeavour for me. If you can sponsor the development of Cachet, I'd love you forever. You can [sponsor me here on GitHub](https://github.com/sponsors/jbrooksuk). I've applied for GitHub Organization Sponsors, but this will do for now.

I guess that's everything for now. Let me know what you think and importantly, whether you can help.

Thanks ❤️

**Comment by vedranmiletic** [comment]
> People who are willing to test Cachet in the future.

I would be interested in this.

**Reply by nixmomo** [reply]
same here, how can we get in touch? 
i can also help with translations into german language if needed

**Reply by taalas** [reply]
Sounds great! Let me know if I can be of assistance!

**Reply by RobertBoes** [reply]
I would be interested in this as well

**Reply by Tensibai** [reply]
Using the 2.4.0-dev branch actually, so interested in testing 3.0 in parallel

**Reply by jbrooksuk** [reply]
There's nothing to test yet, sorry! We're still roadmapping, https://github.com/orgs/cachethq/discussions/4349

**Reply by Tensibai** [reply]
@jbrooksuk that was more a comment to show interest. I've read the roadmaping announcement but it may worth a GitHub project or something to specify these epics and subtasks ?

**Reply by daltonpro** [reply]
This is fantastic news! I'll be watching for the official release.

**Reply by jbrooksuk** [reply]
Thanks @daltonpro!

**Comment by alexjustesen** [comment]
This is awesome news and never too late, I constantly come back to this project as I work in the enterprise and we have the need for self-hosted status pages (re: Atlassian is f***ing expensive).

**Comment by fritz92philipp** [comment]
That's nice 😀🥳
I can help with translation into German & Italian languages and for testing everything 😊

**Reply by BrunoBlanes** [reply]
Me too, translation to Brazilian Portuguese, testing and documenting.

**Reply by nathanaelhoun** [reply]
I can help for French and testing!

**Reply by cuoyot** [reply]
I can help with the Spanish if needed.

**Reply by bugrakeskin** [reply]
i can help with the Turkish...

**Reply by afsharone** [reply]
I can help with the Persian if needed.

**Reply by thl3538** [reply]
I can help with the Chinese if needed

**Comment by w4rc0n** [comment]
Welcome back <3

**Comment by TheDelta** [comment]
This is great news, but sounds like that it will take a while (considering the long wait and silence I guess this doesn't matter).
One thing I would like to see in the future is better handling of communication. While I understand that some things take time and depends on other parties, many users (including me) have been left in the dark and unknown for a long time.

Unfortunately other forks weren't successful either and I hope that 3.x will be a great comeback.
Truly need a nice looking and well working self hosted status page which works fine on latest php versions.

Would love to help but me being an Angular guy and also time is always an issue. But I wish you all the best with your "baby" and hope you can continue where you left and bring it back to live soon. <3

**Reply by jbrooksuk** [reply]
@TheDelta that's absolutely a fair concern, but that won't be the case going forward. The previous owners never communicated that the project had changed hands, and that wasn't my place to do so either.

I've already started rewriting the core of Cachet privately (in preparation for the purchase to complete). I've made a start at:

- Models
- Stubbed Controllers
- Stubbed Http Resources (to replace the current presenters)
- Migrations to bring the database structure up to "modern standards"

There is a lot to do, of course. But progress _is_ being made.

Next, I need to decide what to do with the current issues and pull requests. I'm thinking I'll go through each one, make a list including:

- What's a bug in 2.4 and **must be fixed.**
- What could we bring into 3.x as a new feature, to make the upgrade worthwhile?

By the end of it, everything will be closed and we'll create a new project board or something with a list of the bits that need doing.

**Comment by Sufiyanshaikh123** [comment]
I can test Cachet in future for you

**Reply by jbrooksuk** [reply]
Thanks! :)

**Reply by shiroopi** [reply]
me as well !

**Comment by MarcHagen** [comment]
Very nice to see a good open source project like this being back to live again.
Tho I'm hoping you have a good game plan and design plan for Cachet, just rebuilding the code is not the right way to go.
A new fresh way is better. Learn from 2.4-dev and improve greatly for 3.x. Take the time and please do not rush.

**Reply by jbrooksuk** [reply]
@MarcHagen thanks! 

You're right, there is a lot more to it than simply rebuilding it like-for-like. There's a lot of work, but I'm confident we can do this 💪

**Comment by dovka** [comment]
Great to hear the project is back alive!
One of the best status pages because it allows to present the services in hierarchical fasion :)
Will you keep the Rest API compatability in 3.x ?
Thanks a lot - looking forward to 3.x!

**Reply by jbrooksuk** [reply]
@dovka thank you for your kind words ❤️ 

Cachet will keep its API, though it may not be 100% compatible with the current 2.x API, simply because we can do things much better and offer way better performance now.

**Comment by gautamkrishnar** [comment]
This is awesome, all the best for the project. If you need any help maintaining the project please keep creating Github issues about your plans. Once we have a good base code, all of us can contribute.

**Reply by jbrooksuk** [reply]
Thank you 😄

**Comment by misterjaytee** [comment]
Happy to test and proof-read documentation. 

Moved to another open source status page system because this one wasn't being maintained, then when that stopped being maintained went onto Atlassian (which is quite restrictive on it's customisation unless you go for the $399 per month plan).

Would be great to be able to go back to the first choice again.

Are there plans to do a docker/container release of the new version 3.x?

P.S. The blog link is dead on the Cachet HQ website (cachethq.io)

**Reply by jbrooksuk** [reply]
@misterjaytee thank you!

Cachet really set the bar for open-source status page systems and we're going to take the 👑 back and make it better still. 

I'm sure there will be a Docker container available with 3.x once we launch it.

Also, I've just shipped a new Cachet website. There is no active blog (currently).

**Comment by donalturrentine** [comment]
Wonderful news!!! I would suggest definitely leaving the door open for translations. I suspect there is a large amount of interest globally. I’m happy to help test!

**Reply by jbrooksuk** [reply]
Thanks! Translations are **without a doubt** a huge part of what made Cachet successful the first time round. We won't be completely dropping them, but the existing ones may not be compatible to begin with.

**Comment by MikeKMiller** [comment]
Welcome back!! Happy to help pitch in wherever I can.

**Reply by jbrooksuk** [reply]
Thanks @MikeKMiller 💚

**Comment by YWatchman** [comment]
Welcome back! To start of good with enough cups of coffee, I sponsored the project through your personal page with a $100 funding.

**Reply by jbrooksuk** [reply]
Thank you so much, @YWatchman 💚 Your support means so much.

Let's do this 💪

**Comment by JakeyPrime** [comment]
So happy to see you back, we all missed you so much. Congratulations on having a kid! 

I'll throw a lot of money at you to keep this going as soon as we're able!

**Reply by jbrooksuk** [reply]
Thanks, @JakeyPrime! 💚

**Comment by MatthewSH** [comment]
It's been a long time since I've been in the Laravel community but I've been itching to get back in it in some capacity. 

I still have need for this even to this day and would love to see some of it come to life. 

I'm willing to join in on the development process in any capacity. 

Do you all have a Discord community you are coordinating in? If not, I'll be happy to help set it up since I run Discord communities on the side.

**Reply by jbrooksuk** [reply]
@MatthewSH 👋🏻 

I love that people are still actively using Cachet **and** even new companies need it too. 

We do have a Slack, but I do think Discord may be better suited going forward. For now, I think we'll stick to issues and discussions. Maybe transition to Discord in the future.

**Reply by MatthewSH** [reply]
> @MatthewSH 👋🏻 
> 
> 
> 
> I love that people are still actively using Cachet **and** even new companies need it too. 
> 
> 
> 
> We do have a Slack, but I do think Discord may be better suited going forward. For now, I think we'll stick to issues and discussions. Maybe transition to Discord in the future.

Sounds good! If you need anything in that area, please feel free to reach out!
I'll keep an eye out on the repo other wise to see how it progresses and see if there's anything I could do to help too.

**Comment by Isengo1989** [comment]
Sounds great, I really liked using it back in the day!

I am happy to help with documentation :mechanical_arm:

**Reply by jbrooksuk** [reply]
Thanks @Isengo1989!

**Comment by rpartapsing** [comment]
Still using it daily in production inside our k8s cluster. Happy to help you test the new platform. Nice to have you back!

**Reply by jbrooksuk** [reply]
That's awesome! 🔥

**Comment by ricardov03** [comment]
Ready to work, James! Mostly because I would LOVE to work with you. So, let's go!

**Comment by trianity** [comment]
@jbrooksuk I was happy to read the news that the project is alive again. I am happy to help: testing, translation into Hungarian, backend coding based on Laravel 10, frontend Tailwind CSS, LiveWire, AlpineJs.

**Comment by vikashgounder** [comment]
@jbrooksuk happy to help with testing the product and also with documentation.

**Comment by demlasjr** [comment]
@jbrooksuk nice to see you back! I'm one of those who contributed to cachet 9 years ago and I'm still using version 2.4.0-dev because I simply loved it. I'll do my best to help you!
You made my day with this news!

**Reply by jbrooksuk** [reply]
@demlasjr good to see you around! Thank you so much for sticking with Cachet. 3.x is going to be awesome.

**Reply by demlasjr** [reply]
I have no doubt about that! I constantly checked github for file changes and I never seen any single one since you left from here. I always doubt why this project died and never find a good replacement for this one. I'm using 2.4.0-dev and fixed a lot of translations and things, but it's still buggy and more with so many abandoned laravel packages. I had to update some of them (for example, the last one was file() to dispatch()) to make this work but never gave up LOL.
And really sorry for your loss. Things changes sometimes in worst and we simply don't find enough motivation to get up and step up, it happened and still happens to me too. 
Also, glad to hear that you became a father. Hope you enjoy that part of your life and wish you all the best,

**Comment by Florian-Mahlberg** [comment]
Heyy,
I've been working on a cachet for for the past couple of weeks. I managed to make a [fork](https://github.com/Hostzero-GmbH/Cachet) with laravel 9 and finished SMS notifications. If you want to use that, you can. It still uses some of the old libraries but it works well enough :)

**Reply by jbrooksuk** [reply]
Thanks for your efforts, @Florian-Mahlberg!

**Reply by tomck** [reply]
There have been several forks, and they all seem to get abandoned really quick. I would love to see one get some love, a 2x series and a 3x series.  Of course I think an official fork that had contributors and imported the issues would be best.

**Comment by msa-tech09** [comment]
Hi there,

I'd be happy to contribute to this project. Count me in for frontend support - I'm experienced with TailwindCSS and eager to help. Looking forward to being part of Cachet's journey.

**Comment by m-t-a97** [comment]
I would be happy to contribute 👍. I have experience with Vue.js and TailwindCSS so looking forward to working on the frontend.

A few questions I have:

- I see that the landing site is currently using Vue.js, would you be open to building the site using Nuxt3?


- For the main product, are you still thinking of using Vue.js or would you also consider Nuxt3 as an alternative since it has many useful features such as SSR/SSG etc?

**Reply by jbrooksuk** [reply]
Thanks! I appreciate your support.

1. I have no plans to change the website to Nuxt at this time. We’re using Vue and Inertia as our stack for a few reasons that’ll help with future plans there. I haven’t yet enabled SSR for the website but it will be soon.
2. Again, Vue and Inertia is the stack for Cachet too. We’ll be able to leverage SSR there too, although it’s possible the status page may be built in Blade just for simplicity.

**Comment by mttradebyte** [comment]
Hi @jbrooksuk,

We actually made a fork of fiveai/Cachet and updated it to require PHP 8/8.1 and Laravel 9.x, as well as some other changes here and there. It's in our internal GitLab as is our policy, however if it is of interest I can see if we can get this work merged back into your repo as a basis for further work if you'd accept that.

If not, well, looking forward to what you come up with!

**Reply by Florian-Mahlberg** [reply]
Hey,
my pr got declined https://github.com/cachethq/cachet/pull/4345 so I guess cachet will just get recoded and no old gets touched anymore

> Hey @Florian-Mahlberg, thank you so much for maintaining this fork of Cachet.
> 
> We're not going to merge this now as 2.x is now considered unsupported, while we focus our efforts on 3.x.
> 
> We hope to get the new `cachethq/core` package open-sourced in the next week or so.
> 
> Thanks again!

**Reply by mttradebyte** [reply]
Yeah, that's fair enough, I can understand if he doesn't want anything else merged into it. I'm just offering in case he can build on some of the work we've already done to save him time and effort.

**Reply by jbrooksuk** [reply]
2.x is now no longer supported simply so we can focus our efforts on 3.x. I think this makes the most sense given how dated 2.x. There are numerous forks available if anyone does need Cachet 2.x in the meantime 😄

**Comment by wfjsw** [comment]
Whoah. Welcome back ;D

**Reply by jbrooksuk** [reply]
Thanks!

**Comment by bugoff** [comment]
Nice to hear you're back! made a donation to you so we can get 3.0 released, didn't know if the @cachethq is also linked to your sponsership?

**Reply by jbrooksuk** [reply]
Thank you so much, @bugoff! Yeah, both are linked to the same place at the moment.

**Comment by andreaskviby** [comment]
I would too if you choose Livewire instead of Inertia 😯

**Reply by jbrooksuk** [reply]
Over the weekend I was thinking about using a touch of Livewire on the status page rather than Vue/Inertia. But, I'm keeping the use of Vue and Inertia in the dashboard.

**Comment by tomck** [comment]
Have you got a roadmap in mind? I think issues I run into with 2.4-dev may still influence 3.x, so I want to get ahead of them and influence the process early on.

**Reply by jbrooksuk** [reply]
Hey @tomck 👋🏻 At present, we don't have a public roadmap (though I'm actively working on it). However, we've been diligently focusing on several other important aspects as well:

- [x] [Acquiring Cachet](https://blog.cachethq.io/cachet-3-x-announcement/)
- [x] [Updating the branding](https://blog.cachethq.io/new-version-new-branding/)
- [x] [Revamping the documentation](https://docs.cachethq.io)
- [x] Initiating the `cachethq/core` repository
- [ ] Creating a public roadmap

I've been concentrating on the `cachethq/core` project for a week or so, and this is where our primary efforts will be directed going forward. I'll be open-sourcing it very soon, enabling others to begin contributing. From that point forward, we'll be able to develop a comprehensive roadmap. Hope that helps.

**Reply by ricardov03** [reply]
Thanks for sharing @jbrooksuk. Please use this channel to make the Open Sourcing announcement to be able to follow up on the project—many thanks for considering my request.

**Reply by jbrooksuk** [reply]
We'll be using Discussions going forward, @ricardov03 :)

**Reply by ricardov03** [reply]
Moving to the discussions! Thanks for sharing @jbrooksuk.

**Comment by jbrooksuk** [comment]
We've just released the Cachet Core package! Read more about the release at https://blog.cachethq.io/cachet-core/

**Comment by roblevesque** [comment]
I'm very excited to see this announcement! We use Cachet at my org and we were literally just discussing today what we are going to do with our status page that is running on a now EOL Ubuntu 18.04 box in Azure.  Until seeing this I was concerned that Cachet was abandonware and I would have to find an alternative (for which the field is fairly baron).

Thank you and I will be watching your progress! I hope there is a clear migration path to 3.x from 2.x

**Reply by jbrooksuk** [reply]
You've found us at the right time! https://blog.cachethq.io/cachet-core/

**Comment by Zauxst** [comment]
Moving forward it would be good if the project can somehow move into the cncf court.

While you might want to monetize the system , it would be nice to ensure the app can live on beyond the purposes of the creator. 

Thanks for your time.

**Reply by tomck** [reply]
> While you might want to monetize the system , it would be nice to ensure the app can live on beyond the purposes of the creator. 

You would think the forks would enable that, but it feels like without pushing users to one of the forks, none get momentum.

**Reply by Zauxst** [reply]
The reality is that most of us are integrators of some sort. 

I also haven't seen that many successful forks come to fruition more popular than the main. 

If the maintainer doesn't step up with this initiative sadly we'll see this happen all over again.

**Comment by pull3t** [comment]
Hi,

I use Catchet v2.3. I'd love to be involved in testing of version 3. Let me know how I can assist. Thanks!

**Comment by ashwinsrinivasmurthy** [comment]
I can help in testing; could you count me in? Also, one of the features we would need is providing an HTML tag feature in incident updates.

**Reply by jbrooksuk** [reply]
@ashwinsrinivasmurthy of course!

By HTML tags, do you mean that Incident Updates can't currently support HTML? They do support Markdown, so you can still format text.

**Reply by MikeKMiller** [reply]
I am kinda with Ashwin, Markdown is massively overated, and doesnt come anywhere close to the capabilities of plain html and css

**Reply by jbrooksuk** [reply]
Personally, I think that Markdown is the right format for incidents, updates and scheduled maintenance. With support for basic formatting, images and the option for inline HTML, what is missing?

**Reply by ashwinsrinivasmurthy** [reply]
So James, does Cachet 2.4 support markdown, or will it be available in 3. x? Also, what i was looking is to see a hyperlink an external URL in status updates like click [here](https://cachethq.io/)

**Reply by ashwinsrinivasmurthy** [reply]
Never mind, I figured it out, and this is already available via markdown in 2.4. Thanks and happy to provide validation support.

**Comment by Tensibai** [comment]
For my part I'm with James, allowing html is a deep security risk and I fail to see what html could provide that markdown doesn't in the use case of a maintenance or incident report.Maybe if you could highlight a real use case on what you're arguing for it could be easier to either highlight in the interface how to achieve it in markdown (any html/css option is doable in basic markdown, even bootstrap like styling as long as it doesn't include javascript) or understand what the use case is and how to integrate it.A chapel fight about html-css vs markdown without an use case isn't the way to go IMHO, it needs a real world example not doable with markdown as a starting point user story.Le 19 sept. 2023 15:48, Mike Miller ***@***.***> a écrit :
I am kinda with Ashwin, Markdown is massively overated, and doesnt come anywhere close to the capabilities of plain html and css

**Comment by Chrisx84** [comment]
i can help with API stuff. I  been using Laravel since the 5.0 was released but im not good with tailwind yet, still using bootstrap.

**Comment by tolecnal** [comment]
I am a bit late to the party here, really glad to hear that this project is catching some fresh wind again! :) 

I would be more than happy to offer my assistance and time to lend a helping hand to get this baby on its feet again. While the backend stuff isn't quite my cup of tea, I would be more than happy to help with documentation (in English), and translations to Norwegian - as I do really hope that translations will be kept in the new revamped version.

Another area I can contribute to is testing, Docker-related stuff and ways to interface the new Cachet with PowerShell for those using Windows. Personally, I would also love a role for Ansible to automate the deployment, which I can help with.

**Comment by andrepg** [comment]
There's a lot going on here, but I have some plans to my future and contribute to this project seems to fit it. I can help to document and translating to Brazilian Portuguese, if no one has offered before. Or I can do anything that I'll be asked and could do o/

**Reply by jbrooksuk** [reply]
Thanks @andrepg!

**Comment by thedom4302** [comment]
What is the planned release date/target for the beta?

**Reply by jbrooksuk** [reply]
@thedom4302 early 2024, I'm hoping. There's a lot to do and we're still looking for volunteers :)

**Comment by JSONbored** [comment]
Happy to help with testing! Love the vision and drive here. Starting to dive real deep into self-hosted microservices, and would love to contribute to maintaining and testing Cachet @jbrooksuk.

**Reply by jbrooksuk** [reply]
@gh0stdotexe thanks! I'd love your support too. We really need to crack on with the design implementation.

**Comment by HerrSammyDE** [comment]
Is there a roadmap for when and how something will go online?

**Reply by jbrooksuk** [reply]
No public roadmap at the moment. You can follow progress of the [new core at the repository](https://github.com/cachethq/core).

**Comment by AdoDreamtilt** [comment]
Hi, just wondering if there is any updates on how this project is progressing? :)

**Reply by jbrooksuk** [reply]
Oh yeah, I have a lot I need to share. I'll post a proper update in a new discussion / blog post. 

However, most recently, I've merged the new dashboard, powered by Filament. https://x.com/jbrooksuk/status/1747714985550135329?s=46&t=miVykdMohea9sEzUdQLynw

**Reply by andrepg** [reply]
Personally, I'm waiting to this announcement, so you can let us know where you are and how we can help you. Can be a good replacement to Uptime Robot, which I'm using today.

**Reply by jbrooksuk** [reply]
I've started writing a new blog post that details the latest updates from Cachet. I hope to get it finished this weekend and published for Monday.

I would also recommend that you check out the [`cachethq/core`](/cachethq/core) repository, as that's where all active development is happening. `cachethq/cachet` will become a single application that installs and configures the core in a similar set up to 2.x

**Reply by jbrooksuk** [reply]
@AdoDreamtilt @andrepg here you go, https://blog.cachethq.io/cachet-development-update-jan-2024/

**Comment by cumanzorx07** [comment]
I would like to contribute to the project, if you still need volunteers.

**Reply by jbrooksuk** [reply]
Of course! Please take a look at https://github.com/cachethq/core to see the open issues and PRs.

**Reply by cumanzorx07** [reply]
Ok, I will take a look

**Comment by jeffreybarbosa** [comment]
@jbrooksuk After updating to 2.4, when we create an incident the page generates error 500.

**Reply by jbrooksuk** [reply]
@jeffreybarbosa can you please create an issue at https://github.com/cachethq/cachet/issues - be sure to include any logs to help us identify what is causing that to happen.

**Comment by jaggerwang** [comment]
Does 3.x plan to support webhook? As we want to notify our online users when publish an maintenance on Cachet, it seems 2.x still not support webhook right now.

**Reply by jbrooksuk** [reply]
That wasn't in the initial plan, but I do see the value in this. I'm sure we can implement that.

**Comment by abhishek-vasyerp** [comment]
I've an issue regarding mail, i can't change the mail from address, every time i refresh it, the default address is shown.

**Comment by demlasjr** [comment]
Hi everyone? Isn't 3.x available to install yet? Any ETA? Thanks

**Reply by jbrooksuk** [reply]
Hey! Nope, it's not available yet, but we're still actively working on it. You can see where we're at at https://v3.cachethq.io

By all means, I'd love your contributions!

**Reply by demlasjr** [reply]
Thank you for confirmation James. I tried to install it and had no chance, that's why I asked you. :)

**Reply by AdoDreamtilt** [reply]
How is the project progressing? Would be great to get an update and possible ETA. We are keen to run a trial on our systems here.

**Reply by AdoDreamtilt** [reply]
Hi, following up again - is there a chance for an update on this project please?

**Reply by AdoDreamtilt** [reply]
Hi James, please can we have an update (sponsoring). If there are issues or more help is required, please let us know and provide some details on requirements and we will help. Understand progress issues may occur and there is alot to do, but please keep us updated so we know the project is still progressing and sponsorship still occurring. Happy to assist anyway we can to see this project successful and up and running.

**Reply by jbrooksuk** [reply]
Hey @AdoDreamtilt, sorry for the delay! I was promoted at my job and it's been a busy couple of months while transitioning in that role (and the project we have on).

I've received a couple of emails recently about vulnerabilities in 2.x which I am considering fixing in the short term.

For 3.x, I've also been working on it recently, mostly smaller fixes. https://github.com/cachethq/core

Your offer to help is much appreciated. I wonder if we could have a call at some point to discuss how we could move forward with that?

**Reply by AdoDreamtilt** [reply]
Yep - sounds good - lets try to catch up :)

**Reply by jbrooksuk** [reply]
> Yep - sounds good - lets try to catch up :)

Great! Can you send me an email, james@cachethq.io and we'll get something set up.

**Comment by Chrisx84** [comment]
sooo any update on the progress of 3? is it going to support php 8.3 and laravel 11?

**Reply by jbrooksuk** [reply]
> sooo any update on the progress of 3? is it going to support php 8.3 and laravel 11?

@Chrisx84 yep, quite a few updates recently. Also, we recently published a blog post with some updates https://blog.cachethq.io/cachet-3-x-update/

Work is mostly happening at [`cachethq/core`](/cachethq/core).

**Reply by Chrisx84** [reply]
cool so can i install 3 with just composer create-project ?

**Reply by jbrooksuk** [reply]
@Chrisx84 yep, so long as you use `cachethq/cachet`. If you want to develop Cachet, I'd clone `cachethq/core` and then run `composer start`.

**Comment by golgert123** [comment]
Hello,

I read your conversation with another poster above about having a webhooks feature for 3.x. I was curious, is it 100% in the works, can we expect to see it sometime soon? 

Love the work you and the contributors have done so far, keep it up!

**Reply by jbrooksuk** [reply]
Thanks! I've just created the issue for the webhooks feature in the core repo - https://github.com/cachethq/core/issues/106

**Comment by golgert123** [comment]
Hey,

I think it would be neat to see some sort of SSO SAML / LDAP / IP allowlisting, or any other additional form of authentication.
No worries if it won't be a priority, I understand there are a lot of moving parts at the moment! Let me know what you think about it.

Thanks again

**Reply by jbrooksuk** [reply]
Thanks! This won't be something we target for 3.0, but it'll be something for the future, definitely.

**Comment by TheDelta** [comment]
It has been now over a year and I had to ditch php7.4 support once and for all.

I'm a "nobody user" with a very basic cachet setup, I could migrate to 3.x now with a few issues which I could resolve:

I couldn't find any migration guide and docs are not fully helpful yet, so I took all the pieces I found and put them together.
(Hopefully complete) Instructions at your own risk (have a backup of your DB):

```bash
# cd into cachet
# checkout to 
git checkout .

# I removed vendor and the cache manually (whole env was broken due to only php8.1 - php8.3)
# rm -R vendor/
# rm -R bootstrap/cache

# checkout to current state
git checkout .

# switch to 3.x
git switch 3.x

# NOTE if you come from an older system, you might have to install latest composer and or use a different php version
# run composer
composer install --no-dev -o
# or force a given php version `/usr/bin/php8.2 /usr/local/bin/composer install --no-dev -o`

# Migrate to new .env
# I simply took the .env.example and filled in my old values from the .env
# any obsolete keys were removed
# I chose:
APP_ENV=production
APP_KEY=<oldkey>
APP_DEBUG=false

# also I had mysql and wasn't sure which key was the right one:
DB_CONNECTION=mysql
DB_DRIVER=mysql

# email keys have slightly changed

# clear everything to be sure
php artisan config:clear
php artisan cache:clear

# then run php artisan
php artisan vendor:publish --tag=cachet

# now migration...
# In my case, I had to change the collation of all tables to `utf8mb4_unicode_ci` (it was somethig with utf8mb3 before)
# I renamed the user table, because in my case the migration wants to create a new one
# The settings table migration failed, I had to empty the table
# then run migration (say yes for production)
php artisan migrate

# it should pass
# I then migrated my two users back :)
# And some settings (through DB and frontend)

# If the frontend is not working (lots of errors in console), then try to install / upgrade the filament stuff:
php artisan filament:install
php artisan filament:upgrade
```

Haven't fully tested it yet. But works for my usecase:
I have my components back and a working (very nice frontend <3)
Its not fully working as far as I can tell, unsure if this is due to still in development or something not migrated properly (need to check)

**Reply by jbrooksuk** [reply]
@TheDelta I'm glad to hear you got 3.x working - sorry the docs aren't complete. This is something I _really_ want to get sorted soon. 

> Its not fully working as far as I can tell, unsure if this is due to still in development or something not migrated properly (need to check)

Which bit(s) aren't working for you fully? Metrics are not currently implemented, so you may be missing some of that.

**Reply by TheDelta** [reply]
First of all congrats to what has been achieved so far.
I love the new UI and everything just feels fresh and modern ❤️ 

Yeah I noticed that metrics are not working (just had a test, not used it actively yet) and the logo upload (banner image) doesn't seem to work or I must have something misconfigured (upload seemed to work, it uses some <hashed>.png file but it isn't stored in public folder).
Other non working parts were actually related to missing frontend files, after executing the filament command it worked.
API url changed, the version part was removed.
I also had to finish all planned schedules.

Other than that it went surprisingly well and works for my needs now. 👍

(Edit: hit reply too early 🙄)
Also subscribers isn't fully implemented yet, right?
I receive no mails and I can't seem to register / unregister.

Would love to help somehow but not much time and I think it's better when someone writes the docs who knows what he is talking about :D I can only offer some feedback and some hacky upgrade guide :)

**Reply by jbrooksuk** [reply]
@TheDelta thank you 😊 A huge thanks to Jump24 for the new design!

Yeah, so there are a few things currently unfinished / not implemented including (but not limited to 🙈):

- Metric charts
- Subscribers
- Tags
- Meta

By the way, I've just updated a load of documentation for v3.x which should help you https://docs.cachethq.io/v3.x. In particular, https://docs.cachethq.io/v3.x/configuration/theme should help you get the custom logo working.

**Reply by TheDelta** [reply]
@jbrooksuk thanks for the quick update ❤️  Works perfectly now 🥰 
Updated to latest release version and ran the link command from the documentation.

For the basic setup it works well and is all I need so far, just need to update my check_mk script to work with the v3 API and then I'm happy 😄
Keep it up and looking forward to the release of v3 👀

**Comment by PeterDaveHello** [comment]
Looking forward to Cachet v3! I'm interested in helping with Traditional Chinese translations. I think some of the existing translations could still be valuable and worth considering for reuse in the new version, even if they might need some format adjustments. Also, is it still possible to contribute to 2.4 translations before 3.0 is released? Thanks!

**Reply by jbrooksuk** [reply]
@PeterDaveHello thanks! Translations would be great, but definitely not until we've released v3.0 as things are still changing.

**Comment by jbrooksuk** [comment]
Happy holidays, everyone!

It's been a busy year. I'll soon publish a year in review post for Cachet.

**Comment by stxrlite** [comment]
any idea when v3 will be released?

**Reply by jbrooksuk** [reply]
@stxrlite no hard dates at the moment. We're working hard on v3 though and we expect something by the end of the year. 

https://blog.cachethq.io/2024-recap/

**Reply by gdlxn-ibm** [reply]
From [Cachet v3 Installation](https://docs.cachethq.io/v3.x/installation):

> Cachet v3.x is currently in development and not yet released. The following instructions are intended for development purposes only and are subject to change.

Any update on the Cachet v3 release date?  Is it still targeted for year end?

**Reply by jbrooksuk** [reply]
Nothing targeted, which I'm happy with right now. Work has been incredibly busy (hence my late replies) and I'm focused a lot on that.

I'd love some help to finish off the last bits of Cachet 3.x though.

**Reply by No-incident-Sam** [reply]
Hello, @jbrooksuk i've just read your post of August 2023 and hope you're doing well since then

Congratulation on your daughter birth and the work you do on cachet 3.X. You singlehandly made me create a github account just to follow your project, as we plan to use it within the company i work for once the 3.X will be avalaible. on this matter, do you have any update on a possible release date ?

i am unfortunately not good enough to assist you out on a technical level, but i will be glad to help you on a French translation or any testing phase if you ever need them

**Comment by hanisirfan** [comment]
Hi @jbrooksuk , I've heard about Cachet but never tried it before. Looking at the v3 demo and it's really nice. I'm currently looking around in the open source space for status pages. I've tried Uptime Kuma but unfortunately it doesn't have incidents history like Cachet have. Any idea when v3 will officially be released? Thanks :D

**Reply by jhagan-91** [reply]
> Hi @jbrooksuk , I've heard about Cachet but never tried it before. Looking at the v3 demo and it's really nice. I'm currently looking around in the open source space for status pages. I've tried Uptime Kuma but unfortunately it doesn't have incidents history like Cachet have. Any idea when v3 will officially be released? Thanks :D

Sadly looks like this has yet again been abandoned.

**Reply by enyineer** [reply]
Remember: this is open Source. You can fork it and continue development.

Sure, it's sad because it sounded promising, but there's nothing you can demand from unpayed OSS devs.

**Reply by DennisSkov** [reply]
@moroneyio why do you assume Cachet has been abandoned again?

**Reply by gdlxn-ibm** [reply]
@DennisSkov One concern of mine is that this Cachet 3.x announcement was created on Aug 10, 2023.  It's now close to the end of 2025 with Cachet 3.x still not officially released and no clear date on when it will be released.

**Reply by jbrooksuk** [reply]
We're getting there, sorry! AI has supercharged our ability to work through the backlog.

**Comment by singhangadin** [comment]
When is the release coming, Eagerly waiting for it!

**Reply by demlasjr** [reply]
I guess that the answer is 'never'.

I've gone with Kurma myself.

**Reply by jbrooksuk** [reply]
We're getting there! AI has super-charged my ability to ship things.

**Comment by rossnick** [comment]
I've been following this since a while. I guess now with Claude code, one could vibe-code this in a couple of days.
Anyhow, since a 3 years or so, we settled on using instatus.com for our status page.

**Reply by enyineer** [reply]
That's what I'm currently doing :)

It's basically a very extensible monitoring platform focused on flexibility. You can even write Typescript Scripts and track SLOs with dependency awareness.

If you want to, check it out here: [github.com/enyineer/checkstack](https://github.com/enyineer/checkstack)

It's completely free and self-hosted but still in early BETA and changing quite a lot according to feedback I'm receiving.

I'd love if you try it out and break it to give me some feedback as well!
